pilot/pkg/security/model/authentication_test.go

```diff
@@ -22,12 +22,66 @@ import (
 	"github.com/davecgh/go-spew/spew"
 	core "github.com/envoyproxy/go-control-plane/envoy/config/core/v3"
 	auth "github.com/envoyproxy/go-control-plane/envoy/extensions/transport_sockets/tls/v3"
-
 	"istio.io/istio/pilot/pkg/features"
 	"istio.io/istio/pilot/pkg/model"
 	"istio.io/istio/pilot/pkg/networking/util"
+	//"istio.io/istio/pilot/pkg/security/authz/matcher"
+	matcher "github.com/envoyproxy/go-control-plane/envoy/type/matcher/v3"
+	networking "istio.io/api/networking/v1alpha3"
+
 )
 
+func TestConstructSdsSecretConfigWithCustomUds(t *testing.T) {
+	cases := []struct {
+		serviceAccount string
+		sdsUdsPath     string
+		expected       *auth.SdsSecretConfig
+	}{
+		{
+			serviceAccount: "spiffe://cluster.local/ns/bar/sa/foo",
+			sdsUdsPath:     "/tmp/sdsuds.sock",
+			expected: &auth.SdsSecretConfig{
+				Name: "spiffe://cluster.local/ns/bar/sa/foo",
+				SdsConfig: &core.ConfigSource{
+					InitialFetchTimeout: features.InitialFetchTimeout,
+					ConfigSourceSpecifier: &core.ConfigSource_ApiConfigSource{
+						ApiConfigSource: &core.ApiConfigSource{
+							ApiType: core.ApiConfigSource_GRPC,
+							GrpcServices: []*core.GrpcService{
+								{
+									TargetSpecifier: &core.GrpcService_GoogleGrpc_{
+										GoogleGrpc: &core.GrpcService_GoogleGrpc{
+											TargetUri:  "/tmp/sdsuds.sock",
+											StatPrefix: SDSStatPrefix,
+										},
+									},
+								},
+							},
+							RefreshDelay: nil,
+						},
+					},
+				},
+			},
+		},
+		{
+			serviceAccount: "",
+			sdsUdsPath:     "/tmp/sdsuds.sock",
+			expected:       nil,
+		},
+		{
+			serviceAccount: "spiffe://cluster.local/ns/bar/sa/foo",
+			sdsUdsPath:     "",
+			expected:       nil,
+		},
+	}
+
+	for _, c := range cases {
+		if got := ConstructSdsSecretConfigWithCustomUds(c.serviceAccount, c.sdsUdsPath); !reflect.DeepEqual(got, c.expected) {
+			t.Errorf("ConstructSdsSecretConfig: got(%#v) != want(%#v)\n", got, c.expected)
```

Comment:
match the function name `ConstructSdsSecretConfigWithCustomUds` in the error message